src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -285,8 +285,17 @@ internal bool TryWrite(Utf8JsonWriter writer, T value, JsonSerializerOptions opt
 
         internal bool TryWriteDataExtensionProperty(Utf8JsonWriter writer, T value, JsonSerializerOptions options, ref WriteStack state)
         {
+            Debug.Assert(!(value is null));
```

Comment:
nit: 
```suggestion
            Debug.Assert(value != null);
```

for consistency with the rest of the repo.

Reply:
+1